client/src/components/UsedOnTable/provideUsedOnData.js

```diff
@@ -6,22 +6,77 @@ import { loadUsedOn } from 'state/usedOn/usedOnActions';
 const provideUsedOnData = (UsedOnTable) => {
   class UsedOnDataProvider extends Component {
     componentDidMount() {
-      this.loadUsedOn();
+      this.haveFetchedData = false;
+      this.addIntervals();
+      if (this.props.forceFetch) {
+        this.fetchDataFromEndpoint();
+      }
     }
 
     componentWillReceiveProps(nextProps) {
       if (nextProps.identifier !== this.props.identifier) {
-        this.loadUsedOn(nextProps);
+        this.fetchDataFromEndpoint(nextProps);
       }
     }
 
-    loadUsedOn(props = this.props) {
+    componentDidUpdate() {
+      this.fetchDataFromEndpoint();
+    }
+
+    /**
+     * Watch the 'Used on' tab and when it's "active", fetch data from the server
+     * This means that data is lazy-loaded, rather then loading in the background every time
+     * a file is selected, even if the user never looks at the used on table
+     *
+     * This is done in vanilla JS rather than react because there does not seem to be a simple
+     * way to get reactstrap tabs (which are set via PHP Form schema) to talk to the UsedOnTable JS
+     */
+    addIntervals() {
```

Comment:
I don't like this logic at all. It's very hackish and prone to break. I'll try to think of a better way of handling this.

Reply:
There's a link to a slack convo in the description with an alternative solution, check that and see what you think

Reply:
At first glance, Bernie solution look better. However the underlying problem is that this data should be fetch with GraphQL instead. My view is that `provideUsedOnData` is wrong to begin with and we scrap.

We should do this properly instead of dancing around our previous mistakes.

Reply:
I'm thinking we should have a `TabAware` higher order component that detects if it's inside a tab and provide additional info. Maybe we could use react context to keep track of that.

That's just some back of the envelop idea. Let's chat about that tomorrow.